src/shims/foreign_items.rs

```diff
@@ -528,7 +528,16 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                     )?;
```

Comment:
I think this "bare Miri memory" should have a different memory kind, not `MiriMemoryKind::Rust`.

Reply:
Yea... I was wondering about that. It's cleaner, but is there,much value to it?

Reply:
It's easy to do so I'd say there's no reason not to do it.

Reply:
I added `MiriMemoryKind::Miri` for `miri_alloc` memory.